backend/Application/Api/GraphQL/Network/ClientVersionComparer.cs

```diff
@@ -0,0 +1,48 @@
+namespace Application.Api.GraphQL.Network
+{
+    public class ClientVersionComparer : IComparer<string?>
+    {
+        public int Compare(string? x, string? y)
+        {
+            if (String.IsNullOrEmpty(x) && String.IsNullOrEmpty(y))
+            {
+                return 0;
+            }
+            else if (String.IsNullOrEmpty(x) && !String.IsNullOrEmpty(y))
+            {
+                return -1;
+            }
+            else if (!String.IsNullOrEmpty(x) && String.IsNullOrEmpty(y))
+            {
+                return 1;
+            }
+            else if (!String.IsNullOrEmpty(x) && !String.IsNullOrEmpty(y))
+            {
+                var xParts = x.Split(".");
+                var yParts = y.Split(".");
+
+                if (xParts.Length != yParts.Length)
+                {
+                    return 0;
+                }
```

Comment:
I'm not sure why this is sensible. So version `1.0` will be deemed equal to version `3.1.2`?

Reply:
Why not just do a lexicographic sorting, where a prefix of a string is less than it?

Reply:
Hi @abizjak 
We have fixed the semver parsing and removed the code being used for the same. 
We have assumed that the version will be always semver compatible
https://github.dev/Concordium/concordium-node/blob/59f58d14079b980312740daf2f2233c386ed1d90/concordium-node/src/configuration.rs#L30

Reply:
lexicographic sorting puts 10 before 2

Reply:
Lexicographic sorting of parts.